### PR TITLE
Feature/chef repo updates

### PR DESCRIPTION
**Pull request by evansde77**
@shudgston @appeltel @petevg 

deploy command prototype

### CODE REVIEW

File: src/cirrus/plugins/deployers/chef.py [hunk 1]
```diff
@@ -44,6 +59,110 @@ def deploy(self, opts):
...
         LOGGER.info(opts)
 
+        args = self._read_cirrus_conf()
+        for opt in vars(opts):
+            if (opt in args) and (getattr(opts, arg, None) is not None):
+                args[opt] = opts[opt]
+        args['version'] = self.package_conf.package_version()
+        self._validate_args(args)
+
+        attributes = {x:args['version'] for a in args['attributes']}
```
Comment (shudgston) [inline src/cirrus/plugins/deployers/chef.py]: `a` is not used, and where does `x` come from?

Comment (evansde77) [inline src/cirrus/plugins/deployers/chef.py]: solve for x. Exercise for the reader :)

File: src/cirrus/plugins/deployers/chef.py [hunk 2]
```diff
@@ -44,6 +59,110 @@ def deploy(self, opts):
         LOGGER.info('Chef deployment running...')
         LOGGER.info(opts)
 
+        args = self._read_cirrus_conf()
+        for opt in vars(opts):
+            if (opt in args) and (getattr(opts, arg, None) is not None):
```
Comment (petevg) [inline src/cirrus/plugins/deployers/chef.py]: This will ignore an arg that has been deliberately set to None. That's probably okay, though ...

**Comment by shudgston** [comment]
:+1:

**Comment by petevg** [comment]
LGTM/+1

**Pull request merged**